lib/config_base.rb

```diff
@@ -53,18 +54,22 @@ def get(key)
   end
 
   def set(key, value, options = {auto_save: true} )
-    key = normalize_key(key)
-    ensure_config_exists(key)
+    if deprecated_entries.include?(key)
+      @entries.delete(key)
+    else
+      key = normalize_key(key)
+      ensure_config_exists(key)
+
+      # Check if data type is correct. true and false are not of the same type which makes the check complex
+      if value.class != @entries[key][:value].class &&
+        !(boolean?(value) && boolean?(@entries[key][:value]))
```

Comment:
Use 4 (not 2) spaces for indenting a condition in an `if` statement spanning multiple lines.